extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace Discussions;
+
+class ForumDumper {
+
+	const TABLE_PAGE = 'page';
+	const TABLE_COMMENTS = 'comments_index';
+	const TABLE_REVISION = 'revision';
+	const TABLE_TEXT = 'text';
+	const TABLE_VOTE = 'page_vote';
+
+	const CONTRIBUTOR_TYPE_USER = "user";
+	const CONTRIBUTOR_TYPE_IP = "ip";
+	const CONTRIBUTOR_TYPE_DELETED = "[deleted]";
+	const CONTRIBUTOR_TYPE_UNKNOWN = "[unknown]";
+
+	const REGEXP_MATCH_TITLE = '/<ac_metadata.*title="([^"]*)".*>.*<\/ac_metadata>/';
+
+	// A very loose interpretation of markup favoring false positives for markup.  Match
+	// alphanumerics, anything in a basic URL and punctuation.  If any character in the text
+	// doesn't match, assume there is wiki text and parse it.
+	const REGEXP_MATCH_HAS_MARKUP = '/[^a-zA-Z0-9\.\/:\?&%" ]/';
+
+	const COLUMNS_PAGE = [
+		"page_id",
+		"namespace",
+		"raw_title",
+		"is_redirect",
+		"is_new",
+		"touched",
+		"latest_revision_id",
+		"length",
+		"parent_page_id",
+		"parent_comment_id",
+		"last_child_comment_id",
+		"archived_ind",
+		"deleted_ind",
+		"removed_ind",
+		"locked_ind",
+		"protected_ind",
+		"sticky_ind",
+		"first_revision_id",
+		"last_revision_id",
+		"comment_timestamp"
+	];
+
+	const COLUMNS_REVISION = [
+		"revision_id",
+		"page_id",
+		"page_namespace",
+		"title",
+		"user_type",
+		"user_identifier",
+		"timestamp",
+		"is_minor_edit",
+		"is_deleted",
+		"length",
+		"parent_id",
+		"text_flags",
+		"comment",
+		"raw_content",
+		"content"
+	];
+
+	const COLUMNS_VOTE = [
+		"page_id",
+		"user_identifier",
+		"timestamp",
+	];
+
+	private $pages = [];
+	private $revisions = [];
+	private $votes = [];
+
+	private $dummyTitle;
+	private $parserOptions;
+
+	public function __construct() {
+		$this->dummyTitle = \Title::newFromText( "Dummy" );
+		$this->parserOptions = new \ParserOptions();
+		$this->parserOptions->setEditSection( false );
+	}
+
+	private function getForumNamespaces() {
+		return [
+			NS_WIKIA_FORUM_BOARD,
+			NS_WIKIA_FORUM_BOARD_THREAD
+		];
+	}
+
+	public function addPage( $id, $data ) {
+		$this->pages[$id] = $data;
+	}
+
+	public function addRevision( $data ) {
+		$this->revisions[] = $data;
+	}
+
+	public function addVote( $data ) {
+		$this->votes[] = $data;
+	}
+
+	public function getPages() {
+		if ( !empty( $this->pages ) ) {
+			return $this->pages;
+		}
+
+		$dbh = wfGetDB( DB_SLAVE );
+		( new \WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::TABLE_PAGE )
+			->LEFT_JOIN( self::TABLE_COMMENTS )->ON( 'page_id', 'comment_id' )
+			->WHERE( 'page_namespace' )->IN( $this->getForumNamespaces() )
+			->runLoop( $dbh, function ( &$pages, $row ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$pages". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)